File: projectNew/old_src/TTA/TTAAttackResolutionListener.java
package com.f14.TTA.listener;

import com.f14.TTA.TTAPlayer;
import com.f14.TTA.component.card.AttackCard;
import com.f14.bg.common.ParamSet;

/**
 * TTA的攻击结算监听器
 *
 * @author F14eagle
 */
public abstract class TTAAttackResolutionListener extends TTAInterruptListener {
    protected AttackCard attackCard;
    protected TTAPlayer winner;
    protected TTAPlayer loser;
    protected ParamSet warParam;

    /**
     * 构造函数
     *
     * @param trigPlayer
     * @param winner
     * @param loser
     * @param warParam
     */
    public TTAAttackResolutionListener(AttackCard attackCard, TTAPlayer trigPlayer, TTAPlayer winner, TTAPlayer loser,
                                       ParamSet warParam) {
        super(trigPlayer);
        this.attackCard = attackCard;
        this.winner = winner;
        this.loser = loser;
        this.warParam = warParam;
        if (this.attackCard.loserEffect.winnerSelect) {
            this.addListeningPlayer(winner);
        } else {
            this.addListeningPlayer(loser);
        }
    }
}

File: projectNew/src/com/f14/TTA/listener/TTAAttackResolutionListener.kt
package com.f14.TTA.listener

import com.f14.TTA.TTAGameMode
import com.f14.TTA.TTAPlayer
import com.f14.TTA.component.card.AttackCard
import com.f14.bg.common.ParamSet

/**
 * Created by 吹风奈奈 on 2017/7/20.
 */

abstract class TTAAttackResolutionListener(gameMode: TTAGameMode, trigPlayer: TTAPlayer, protected var attackCard: AttackCard, protected var winner: TTAPlayer, protected var loser: TTAPlayer, protected var warParam: ParamSet) : TTAInterruptListener(gameMode, trigPlayer) {

    init {
        if (this.attackCard.loserEffect.isWinnerSelect) {
            this.addListeningPlayer(winner)
        } else {
            this.addListeningPlayer(loser)
        }
    }
}
